### PR TITLE
add: follow feature and user profile

### PR DESCRIPTION
**Pull request by ApurvaSawant11**
In this PR, I have added the below features:
- Follow or Unfollow the user
- User profile page
- Edit user details feature

### CODE REVIEW

**Review by Dev-AkshatGupta — COMMENTED** [review]
Eveything seems cool to me you can go ahead and merge it

File: src/features/userProfile/UserProfile.jsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+  useEffect(() => {
+    user.userHandle === userHandle && setSingleUser(user);
+  }, [user]);
+
+  const isFollowing = selectedUser?.followers?.some(
+    (follower) => follower.username === user.username
+  );
+
+  return selectedUser?.username ? (
+    <div className="min-h-screen py-12 w-11/12 xs:w-4/5 md:w-11/12 lg:w-4/5 xl:w-3/5 m-auto flex">
```
Comment (aayushjain-code) [inline src/features/userProfile/UserProfile.jsx:58]: Class names are bit complex and not easy to understand you can give a bit more understandable names,

File: src/features/userProfile/UserProfile.jsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+                    <div className="flex items-center gap-1">
+                      <UnfollowIcon size={20} /> Unfollow
+                    </div>
+                  )}
+                </button>
+              )}
+            </div>
+
+            <div className="flex text-gray-500 font-semibold gap-8 mb-2 ">
+              <span>{selectedUser.followers.length} Followers</span>
```
Comment (aayushjain-code) [inline src/features/userProfile/UserProfile.jsx:109]: you can destructure all props at once

**Review by aayushjain-code — COMMENTED** [review]
Overall code looks clean and modular you can merge it!

File: src/features/userProfile/UserProfile.jsx (hunk 1 above)
Comment (ApurvaSawant11) [reply src/features/userProfile/UserProfile.jsx:58]: I am using tailwind CSS.  These are their classnames

**Pull request merged**